xea-web/src/components/login/LogInAndOut.jsx

```diff
@@ -0,0 +1,117 @@
+import React, { Component, PropTypes } from 'react'
+import { connect } from 'react-redux'
+import { Modal } from 'react-bootstrap'
+import { loginRequest, logoutRequest } from '../../actions/authActions'
+
+import LoginForm from './LoginForm'
+import UserMenu from './UserMenu'
+
+class LogInAndOut extends Component {
+  constructor (props) {
+    super(props)
+
+    this.state = { showModal: false }
+
+    this.open = this.open.bind(this)
+    this.close = this.close.bind(this)
```

Comment:
Agora mesmo ao constructor tres que mirar igual a ver se se fixo o bind a this, pero tes razón, non hai maneira que non teña defectos. Podemos ir coa túa :)